### PR TITLE
Added new features and fixed bugs

### PR DESCRIPTION
**Pull request by Andre3567**

### CODE REVIEW

File: player/src/main/java/com/tritondigital/player/StationPlayer.java [hunk 1]
```diff
@@ -188,6 +275,30 @@ private void cancelConnectionClient() {
         @Override
         public void onStationConnectionError(StationConnectionClient src, int errorCode) {
             setErrorState(errorCode);
+
+            //Google analytics  :track connection time
+            AnalyticsTracker tracker = AnalyticsTracker.getTracker(getContext());
```
Comment (EddieGr) [inline player/src/main/java/com/tritondigital/player/StationPlayer.java]: Check to remove Analytics Tracker in this file.

Comment (EddieGr) [reply player/src/main/java/com/tritondigital/player/StationPlayer.java]: And everything related to it.

Comment (EddieGr) [reply player/src/main/java/com/tritondigital/player/StationPlayer.java]: Remove whole section

**Review by EddieGr — CHANGES_REQUESTED** [review]

**Review by EddieGr — APPROVED** [review]

**Review by EddieGr — CHANGES_REQUESTED** [review]
Check if you can get to a startup.

File: player/src/main/java/com/tritondigital/player/CuePointHistory.java [hunk 2]
```diff
@@ -108,8 +66,10 @@ public interface CuePointHistoryListener {
     private static final String TAG = Log.makeTag("CuePointHistory");
     private static final int MIN_REQUEST_TIME   = 15000;
 
-    private static final String SERVER_PROD    = "https://np.tritondigital.com";
-    private static final String SERVER_HTTPS   = "https://np.tritondigital.com";
+    private static final String SERVER_PROD    = "https://np.example.com";
+    private static final String SERVER_HTTPS   = "https://np.example.com";
+    private static final String SERVER_PREPROD = "https://example.preprod01.net";
```
Comment (jd-buys) [inline player/src/main/java/com/tritondigital/player/CuePointHistory.java]: Please remove the preprod environment stuff

File: player/src/main/java/com/tritondigital/player/Provisioning.java [hunk 3]
```diff
@@ -65,10 +66,12 @@ abstract class Server
         }
     }
 
-    private static final String DOMAIN_NAME_PROD = "playerservices.streamtheworld.com";
+    private static final String DOMAIN_NAME_PROD = "exampleProdserver.com";
```
Comment (jd-buys) [inline player/src/main/java/com/tritondigital/player/Provisioning.java]: This is the wrong prod domain name. The playerservices.streamtheworld.com was correct.

File: player/src/main/java/com/tritondigital/player/Provisioning.java [hunk 4]
```diff
@@ -65,10 +66,12 @@ abstract class Server
...
+    private static final String SERVER_DEV       = "https://example.Dev.server/api/livestream";
+    private static final String SERVER_PREPROD   = "https://example.Preprod.server/api/livestream";
```
Comment (jd-buys) [inline player/src/main/java/com/tritondigital/player/Provisioning.java]: Remove the dev and preprod values

File: player/src/main/java/com/tritondigital/player/Provisioning.java [hunk 5]
```diff
@@ -176,7 +187,9 @@ private static String createUrl(String mount, String suffix, String userAgent, S
 
         if (suffix != null) {
             switch (suffix) {
+                case ".preprod": serverUrl = SERVER_PREPROD; break;
```
Comment (jd-buys) [inline player/src/main/java/com/tritondigital/player/Provisioning.java]: Remove preprod and dev

File: player/src/main/java/com/tritondigital/player/StationPlayer.java [hunk 6]
```diff
@@ -48,7 +58,12 @@ public class StationPlayer extends MediaPlayer
     private String                  mUserAgent;
     private MediaRouter.RouteInfo   mMediaRoute;
     private String                  mLiveStreamingUrl;
-    private boolean                 timeshiftEnabled = false;
+    private boolean                 timeshiftStreaming = false;
+
+    private int originalSeekValue;
+
+    private static final String DOMAIN_NAME_PROD = "example.com";
```
Comment (jd-buys) [inline player/src/main/java/com/tritondigital/player/StationPlayer.java:65]: The domain_name_prod has the wrong value.
Please remove preprod.

File: player/src/main/java/com/tritondigital/player/StationPlayer.java [hunk 7]
```diff
@@ -104,7 +135,65 @@ protected void internalRelease() {
...
+    public void getCloudStreamInfoAndNotify(){
+        RequestQueue queue = Volley.newRequestQueue(getContext());
+        String mount = getSettings().getString(SETTINGS_STATION_MOUNT);
+        String programUrl = String.format("https://%s/api/cloud-redirect/%s/stream-info", DOMAIN_NAME_PROD, mount);
 
+        try {
+            if (mount.indexOf('.') != -1) {
+                String[] mountArgs = mount.split("\\.");
+                switch (mountArgs[1].toLowerCase()) {
+                    case "preprod": programUrl = String.format("https://%s/api/cloud-redirect/%s/stream-info", DOMAIN_NAME_PREPROD, mountArgs[0]); break;
```
Comment (jd-buys) [inline player/src/main/java/com/tritondigital/player/StationPlayer.java:174]: Remove preprod

**Review by jd-buys — CHANGES_REQUESTED** [review]

**Review by EddieGr — APPROVED** [review]

**Pull request merged**